Changed the term master to main

pr0PM opened the pull request:
Since we are updating it everywhere :)

<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.

See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.

For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).

We cannot accept changes to our translated content right now. See the [contributing.md](/main/CONTRIBUTING.md#earth_asia-translations) for more information.

Thanks again!
-->

### Why: master -> main 

<!-- 
- If there's an existing issue for your change, please link to it.
- If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

### What's being changed: Glossary 

The glossary section has been updated in all the relevant places that came under my eyes while reading it.

<!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. -->

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

pr0PM committed:
Changed the the term master to main

Since we are updating it everywhere :)

janiceilene commented:
Thanks so much for the PR @pr0PM! I'll get this triaged for review 🌟

hubwriter commented on data/glossaries/external.yml:
```diff
@@ -395,10 +395,10 @@
   description: >-
     The default development branch. Whenever you create a Git repository, a
-    branch named "master" is created, and becomes the active branch. In most
+    branch named "main" is created, and becomes the active branch. In most
```
> Whenever you create a Git repository, a branch named "master" is created

Unfortunately this isn't true. If you create a directory, go into it, and do `git init` to create a new repository, in most cases you'll find yourself in a branch called `master` - unless you have configured a git template that changes this to something else, like `main`.

I'd suggest we simplify the definition of `main` here to just:

> Typically, the default branch for repositories on GitHub is called `main`.

pr0PM replied:
So shall I make a commit with proposal to add separate definition of `main` in the glossary as suggested by you, keeping the definition of `master` intact since it's the current standard in Git?

hubwriter replied:
Good point. I think you're right: we shouldn't do away with `master` in the glossary all together. Could you commit another change that alters the definition for `main` and adds an entry for `master`. I think we should simplify the definition of `master` and indicate that it's not a term used at GitHub. How about:

> The default branch in many Git repositories. By default, when you create a new Git repository on the command line a branch called `master` is created. Many tools now use an alternative name for the default branch. For example, when you create a new repository on GitHub the default branch is called `main`.

pr0PM replied:
done

hubwriter reviewed (changes requested):
Thanks for pushing this pull request. We should definitely update the glossary to use `main`. I've suggested a change to the definition for `main`.

janiceilene commented:
@pr0PM If you want to make the changes suggested in this [comment](https://github.com/github/docs/pull/722#discussion_r512120574) and let us know when they're done, we can get this PR merged! Thank you 💝

pr0PM committed:
Updates from the review

Added description for master along with main, since master is still the default on git itself.

hubwriter committed:
Merge branch 'main' into patch-1

hubwriter reviewed (approved):
@pr0PM - thanks. I'll merge this now.

hubwriter committed:
Force tests to rerun

hubwriter committed:
Fix over-indentation

hubwriter committed:
Force test to run again
